jax/_src/cache_key.py

```diff
@@ -129,27 +132,44 @@ def _log_cache_key_hash(hash_obj, last_serialized: str, hashfn):
     )
 
 
-def _serialize_ir(m: ir.Module) -> bytes:
+def _serialize_ir(m: ir.Module, remove_custom_partitioning_ptr_for_cache_key: bool) -> bytes:
   output = io.BytesIO()
   m.operation.write_bytecode(file=output)
-  return output.getvalue()
-
-
-def _canonicalize_ir(m_original: ir.Module) -> bytes:
+  
+  str_m = str(m)
```

Comment:
Could you please avoid string manipulation here?

Here is what I am imagining:
```python
def _replace_unstable_custom_partitioning_configs(m_original: ir.Module):
  def replacer(op: ir.Operation) -> ir.WalkResult:
    if (op.name == "stablehlo.custom_call" and 
        op.attributes["call_target_name"].value == "CustomSPMDPartitioning"):
      op.attributes["backend_config"] = ir.StringAttr.get("<REMOVED_FOR_CACHE>")
    return ir.WalkResult.ADVANCE
  m = type_cast(ir.Module, m_original.operation.clone())
  m.operation.walk(replacer)
  return m
    
def _serialize_ir(m: ir.Module, remove_custom_partitioning_ptr_for_cache_key: bool) -> bytes:
  if remove_custom_partitioning_ptr_for_cache_key:
    m = _replace_unstable_custom_partitioning_configs(m)
  ...
```